mythril/analysis/solver.py

```diff
@@ -122,12 +125,59 @@ def get_transaction_sequence(
         ).as_long()
 
     concrete_initial_state = _get_concrete_state(initial_accounts, min_price_dict)
-
+    if isinstance(transaction_sequence[0], ContractCreationTransaction):
+        code = transaction_sequence[0].code
+        _replace_with_actual_sha(concrete_transactions, model, code)
+    else:
+        _replace_with_actual_sha(concrete_transactions, model)
     steps = {"initialState": concrete_initial_state, "steps": concrete_transactions}
 
     return steps
 
 
+def _replace_with_actual_sha(
+    concrete_transactions: List[Dict[str, str]], model: z3.Model, code=None
+):
+    for tx in concrete_transactions:
+        if hash_matcher not in tx["input"]:
+            continue
+        if code is not None and code.bytecode in tx["input"]:
+            s_index = len(code.bytecode) + 10
```

Comment:
```suggestion
            s_index = len(code.bytecode) + 2
```
I'm assuming this clause is for contract creation transactions where the bytecode is prepended before the input parameters.
In this case I believe the s_index should only be incremented with 2, because there is no function identifier.